posydon/utils/common_functions.py

```diff
@@ -228,6 +228,20 @@ def roche_lobe_radius(q, a_orb=1):
     .. [1] Eggleton, P. P. 1983, ApJ, 268, 368
 
     """
+
+    if isinstance(a_orb, np.ndarray):
```

Comment:
Shouldn't we better check whether a variable is not a `float`?

Reply:
Checking if m1 or m2 are floats causes problems because sometimes they are passed to the function as np.ndarray and sometimes they are np.float64. If we restrict to only floats then integration fails in the detached step later on.

Reply:
If we want to be strict about keeping m1, m2, and a_orb to float types only, then more changes need to be made in other places in the code, because currently they all have inconsistent data types.

Reply:
Or, we can expand the function to allow for floats and arrays of floats, and then change the type checking accordingly

Reply:
Yes, we can take floats and numpy-arrays of floats, but this needs then to be represented in:
- the checking (not included)
- the docstring (states to allow floats only)

Reply:
Currently a mixture of data types are being passed to this function, and the calculation of RL is handling the mixed-types accordingly.